ivy/functional/ivy/nest.py

```diff
@@ -163,12 +170,30 @@ def set_nest_at_index(
         b: ivy.array([3., 4.])
     }
     """
+    outer_tupled = False
+    if _result is None:
+        if shallow:
+            _result = type(nest)(nest)
+        else:
+            _result = copy_nest(nest)
+        if isinstance(nest, tuple):
+            outer_tupled = True
+            _result = list(_result)
     if len(index) == 1:
-        nest[index[0]] = value
+        if shallow:
+            nest[index[0]] = value
+        _result[index[0]] = value
     else:
-        ret = nest[index[0]]
-        ivy.set_nest_at_index(ret, index[1:], value)
-        nest[index[0]] = ret
+        inner_tupled = False
+        if isinstance(_result[index[0]], tuple):
+            _result[index[0]] = list(_result[index[0]])
+            inner_tupled = True
+        set_nest_at_index(nest[index[0]], index[1:], value, shallow, _result[index[0]])
+        if inner_tupled:
+            _result[index[0]] = tuple(_result[index[0]])
+    if outer_tupled:
+        _result = tuple(_result)
```

Comment:
this would modify subclasses of tuple incorrectly, imo we should store `nest_type = type(nest)` instead of `outer_tupled` and simply do `_result = nest_type(_result)` without the `if`, given that if it's not a tuple subclass this won't affect the other classes, don't know if this makes sense